ragna/core/_rag.py

```diff
@@ -148,13 +153,19 @@ def __init__(
         documents: Iterable[Any],
         source_storage: Union[Type[SourceStorage], SourceStorage],
         assistant: Union[Type[Assistant], Assistant],
+        embedding_model: Union[Type[EmbeddingModel], EmbeddingModel],
         **params: Any,
     ) -> None:
         self._rag = rag
 
+        if embedding_model is None and issubclass(source_storage.__ragna_input_type__, Embedding):
+            raise RagnaException
+
+        self.embedding_model = cast(EmbeddingModel, self._rag._load_component(embedding_model))
+
         self.documents = self._parse_documents(documents)
         self.source_storage = cast(
-            SourceStorage, self._rag._load_component(source_storage)
+            VectorDatabaseSourceStorage, self._rag._load_component(source_storage)
```

Comment:
The VDB class is private

```suggestion
            SourceStorage, self._rag._load_component(source_storage)
```